### PR TITLE
Provide just install and solid default

### PR DESCRIPTION
**Pull request by jonhoo**
This removes _most_ of the templates from the repository, and encourages anyone who cannot use the (very opinionated and inflexible) `default.yml` template to write their own `azure-pipelines.yml` (probably by copy-pasting from `default.yml` and hand-tuning) that's based only on the provided `install-rust.yml` template.

This frees us up from all the pain of trying to maintain configuration parameters for every possible use-case, which still allowing easy setup for _most_ projects.

I'd like to hear opinions on this change before merging though.

### CODE REVIEW

**Comment by jonhoo** [comment]
This approach should also result in _much_ faster builds, since we'll no longer be compiling the thing again and again in different jobs.

File: default.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+     - script: cargo check --all --bins --examples
+       displayName: cargo check
+     - script: cargo check --all --bins --examples --no-default-features
+       displayName: cargo check --no-default-features
+     - script: cargo check --all --bins --examples --all-features
+       displayName: cargo check --all-features
+     - script: cargo test --all
+       displayName: cargo test
+       env:
+         ${{ insert }}: ${{ parameters.env }}
```
Comment (epage) [inline default.yml]: If we're not testing under all feature combinations, we should probably include `--tests` under `cargo check`.

Comment (jonhoo) [reply default.yml]: Ah, that was a mistake, I intended for the default to be testing with all features.

**Comment by jonhoo** [comment]
@epage Yeah, the intention behind `default.yml` is that it is for "standard-looking" crates. That is, no nightly-only features, no weird build steps, single crate, etc. That should, I think, cover like 80% of projects. For anything else, the strategy should be to just copy-paste `default.yml` (which isn't really that complicated any more) and modify as needed. Part of the reasoning here is that for the remaining 20% of projects, we end up with a plethora of different, subtly different requirements, and we end up in "parameter hell" like we saw in 0.2 :)

**Comment by jonhoo** [comment]
Consolidated the docs into `README.md` (which I think makes sense now that the repo is simpler) and updated them!

**Pull request merged**